stl/inc/span

```diff
@@ -0,0 +1,732 @@
+        return _Tmp;
+    }
+
+    constexpr void _Verify_offset(const difference_type _Off) const noexcept {
+#if _ITERATOR_DEBUG_LEVEL >= 1
+        if (_Off != 0) {
+            _STL_VERIFY(_Mydata, "cannot seek value-initialized span iterator");
+        }
+
+        if (_Off < 0) {
+#pragma warning(suppress : 4146) // unary minus operator applied to unsigned type, result still unsigned
+            _STL_VERIFY(_Myoff >= -static_cast<size_t>(_Off), "cannot seek span iterator before begin");
+        }
+
+        if (_Off > 0) {
+            _STL_VERIFY(_Mysize - _Myoff >= static_cast<size_t>(_Off), "cannot seek span iterator after end");
+        }
+#else // ^^^ _ITERATOR_DEBUG_LEVEL >= 1 ^^^ // vvv _ITERATOR_DEBUG_LEVEL == 0 vvv
+        (void) _Off;
+#endif // _ITERATOR_DEBUG_LEVEL >= 1
+    }
+
+    constexpr _Span_iterator& operator+=(const difference_type _Off) noexcept {
+#if _ITERATOR_DEBUG_LEVEL >= 1
+        _Verify_offset(_Off);
+        _Myoff += _Off;
+#else // ^^^ _ITERATOR_DEBUG_LEVEL >= 1 ^^^ // vvv _ITERATOR_DEBUG_LEVEL == 0 vvv
+        _Myptr += _Off;
+#endif // _ITERATOR_DEBUG_LEVEL
+
+        return *this;
+    }
+
+    _NODISCARD constexpr _Span_iterator operator+(const difference_type _Off) const noexcept {
+        _Span_iterator _Tmp{*this};
+        _Tmp += _Off;
+        return _Tmp;
+    }
+
+    constexpr _Span_iterator& operator-=(const difference_type _Off) noexcept {
+#if _ITERATOR_DEBUG_LEVEL >= 1
+        if (_Off != 0) {
+            _STL_VERIFY(_Mydata, "cannot seek value-initialized span iterator");
+        }
+
+        if (_Off > 0) {
+            _STL_VERIFY(_Myoff >= static_cast<size_t>(_Off), "cannot seek span iterator before begin");
+        }
+
+        if (_Off < 0) {
+#pragma warning(suppress : 4146) // unary minus operator applied to unsigned type, result still unsigned
+            _STL_VERIFY(_Mysize - _Myoff >= -static_cast<size_t>(_Off), "cannot seek span iterator after end");
+        }
+
+        _Myoff -= _Off;
+#else // ^^^ _ITERATOR_DEBUG_LEVEL >= 1 ^^^ // vvv _ITERATOR_DEBUG_LEVEL == 0 vvv
+        _Myptr -= _Off;
+#endif // _ITERATOR_DEBUG_LEVEL
+
+        return *this;
+    }
+
+    _NODISCARD constexpr _Span_iterator operator-(const difference_type _Off) const noexcept {
+        _Span_iterator _Tmp{*this};
+        _Tmp -= _Off;
+        return _Tmp;
+    }
+
+    _NODISCARD constexpr difference_type operator-(const _Span_iterator& _Right) const noexcept {
+#if _ITERATOR_DEBUG_LEVEL >= 1
+        _STL_VERIFY(
+            _Mydata == _Right._Mydata && _Mysize == _Right._Mysize, "cannot subtract incompatible span iterators");
+        return static_cast<difference_type>(_Myoff - _Right._Myoff);
+#else // ^^^ _ITERATOR_DEBUG_LEVEL >= 1 ^^^ // vvv _ITERATOR_DEBUG_LEVEL == 0 vvv
+        return _Myptr - _Right._Myptr;
+#endif // _ITERATOR_DEBUG_LEVEL
+    }
+
+    _NODISCARD constexpr reference operator[](const difference_type _Off) const noexcept {
+        return *(*this + _Off);

```

Comment:
I am unsure I can follow you here. Do you mean that we should explicitly open code this where it is used and simply retain that function?

Reply:
I mean that `operator[]` is calling `operator+` and `operator*`, when instead it could performing checking and pointer arithmetic directly.

But on second thought, `operator[]` for an *iterator* is rarely used, so no change requested after all.